### PR TITLE
OCPQE-19957: install azure upi with minimal permission

### PR DESCRIPTION
**Pull request by jinyunma**
automate case: OCP-60348 [upi-on-azure] install and destroy cluster with minimum permissions

change jobs from azure-upi to azure-upi-minimal-permission on 4.14+ to cover case OCP-60348

Notes:
1.  The permission `Microsoft.Storage/storageAccounts/blobServices/containers/read` is required when executing command `az storage container exists`, this command is only present in CI step, but not in official document, so I think there is no need to add this permission to doc.

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

File: ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh [hunk 1]
```diff
@@ -192,6 +192,127 @@ required_permissions="""
 \"Microsoft.Storage/storageAccounts/listKeys/action\"
 """
 
+if [[ "${ENABLE_MIN_PERMISSOIN_FOR_UPI}" == "true" ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh:195]: It is possible to get the permission list by `required_permissions` + the difference between IPI and UPI?

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure periodic-ci-openshift-verification-tests-master-installer-rehearse-4.16-installer-rehearse-azure

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.16.yaml [hunk 2]
```diff
@@ -65,8 +65,7 @@ tests:
     cluster_profile: azure-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
-      ENABLE_DES_DEFAULT_MACHINE: "true"
-    workflow: cucushift-installer-rehearse-azure-ipi-disk-encryption-set-minimal-permission
+    workflow: cucushift-installer-rehearse-azure-ipi-minimal-permission
```
Comment (gpei) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.16.yaml:68]: it's configured as ipi-minimal-permission workflow here

Comment (jinyunma) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.16.yaml:68]: yes, I want to verify ipi-minimal-permission also works well after the change in step `azure-provision-service-principal-minimal-permission`.

But [job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49747/rehearse-49747-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.16-installer-rehearse-azure/1767516821574914048) failed as API is unavailable after bootstrap is removed, similar as bug OCPBUGS-29429.
```
W0312 12:35:39.943973     210 reflector.go:539] k8s.io/client-go/tools/watch/informerwatcher.go:146: failed to list *v1.ClusterVersion: Get "https://api.ci-op-9dv0mywm-df3f3.qe.azure.devcluster.openshift.com:6443/apis/config.openshift.io/v1/clusterversions?fieldSelector=metadata.name%3Dversion&limit=500&resourceVersion=0": dial tcp 52.162.255.183:6443: i/o timeout
```

File: ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh]: updated

**Comment by jinyunma** [comment]
[azure-upi-minimal-permission job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49747/rehearse-49747-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure/1767516821503610880)

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.16-installer-rehearse-azure

**Comment by jinyunma** [comment]
azure-upi-minimal-permission: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49747/rehearse-49747-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure/1767516821503610880
azure-ipi-minimal-permission: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49747/rehearse-49747-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.16-installer-rehearse-azure/1767725048384196608

@jianlinliu @gpei pls help to review again, thanks!

**Comment by gpei** [comment]
lgtm

**Comment by jinyunma** [comment]
/pj-rehearse ack

cc @liangxia to review

**Pull request merged**